relayer/relays/beacon/header/header.go

```diff
@@ -555,7 +555,7 @@ func (h *Header) findLatestCheckPoint(slot uint64) (state.FinalizedHeader, error
 	endIndex := startIndex + 1
```

Comment:
Btw: The CI tests failed for running too long,  what about `startIndex` is the last item of the ring buffer, will it cause the endless loop?

Reply:
You're right! Fix here :) https://github.com/Snowfork/snowbridge/pull/1304